.github/workflows/container-image.yml

```diff
@@ -39,202 +37,155 @@ jobs:
-    - name: Download server image
-      uses: ishworkh/docker-image-artifact-download@v1
-      with:
-        image: "samba-server:centos-latest"
-    - name: Test the server image
-      run: LOCAL_TAG=samba-server:centos-latest tests/test-samba-container.sh
-
   build-ad-server:
+    strategy:
+      matrix:
+        os: [centos, fedora, opensuse]
+        arch: [amd64]
+        exclude:
+          - os: centos
     runs-on: ubuntu-latest
     env:
       BUILDAH_FORMAT: oci
     steps:
     - uses: actions/checkout@v3
     - name: Build the ad server image
-      run: make build-ad-server
+      run: make OS_NAME=${{matrix.os}} BUILD_ARCH=${{matrix.arch}} build-ad-server
     - name: Upload ad server image
       uses: ishworkh/docker-image-artifact-upload@v1
       with:
-        image: "samba-ad-server:latest"
+        image: "samba-ad-server:${{ matrix.os}}-latest"
         retention_days: 1
-  build-client-centos:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v3
-    - name: Build the centos client image
-      run: make OS_NAME=centos build-client
-    # Here we upload samba-client image to artifacts locally for consumption
-    # during samba-toolbox build process.
-    - name: Upload client image
-      uses: ishworkh/docker-image-artifact-upload@v1
-      with:
-        image: "quay.io/samba.org/samba-client:centos-latest"
-        retention_days: 1
-
-  build-toolbox-centos:
-    needs: build-client-centos
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v3
-    # Download locally stored samba-client image to be used as base for building
-    # samba-toolbox.
-    - name: Download client image
-      uses: ishworkh/docker-image-artifact-download@v1
-      with:
-        image: "quay.io/samba.org/samba-client:centos-latest"
-    - name: Build the toolbox image
-      run: make OS_NAME=centos build-toolbox
-
 
   build-client:
+    strategy:
+      matrix:
+        os: [centos, fedora, opensuse]
+        arch: [amd64]
     runs-on: ubuntu-latest
-    env:
-      BUILDAH_FORMAT: oci
     steps:
     - uses: actions/checkout@v3
-    - name: Build the client image
-      run: make build-client
+    - name: build the client image
+      run: make OS_NAME=${{ matrix.os }} BUILD_ARCH=${{ matrix.arch }} build-client
     # Here we upload samba-client image to artifacts locally for consumption
-    # during samba-toolbox build process.
-    - name: Upload client image
+    # during  the samba-toolbox build process.
+    - name: Upload the client image
       uses: ishworkh/docker-image-artifact-upload@v1
       with:
-        image: "quay.io/samba.org/samba-client:latest"
+        image: "quay.io/samba.org/samba-client:${{ matrix.os }}-latest"
         retention_days: 1
 
   build-toolbox:
+    strategy:
+      matrix:
+        os: [centos, fedora, opensuse]
+        arch: [amd64]
     needs: build-client
     runs-on: ubuntu-latest
-    env:
-      BUILDAH_FORMAT: oci
```

Comment:
For consistency we could remove above environment variable setting from the entire YAML.

Reply:
I can still see that the environment variable getting removed.